types/propEq.d.ts

```diff
@@ -1,6 +1,31 @@
-export function propEq<T>(val: T): {
-  <K extends PropertyKey>(name: K): (obj: Record<K, T>) => boolean;
-  <K extends PropertyKey>(name: K, obj: Record<K, T>): boolean;
+import { Placeholder } from 'ramda';
+
+export function propEq(__: Placeholder): never;
+export function propEq<V>(val: V): {
+  <K extends PropertyKey>(name: K): (obj: Record<K, any>) => boolean;
+  <U extends Record<PropertyKey, any>>(__: Placeholder, obj: U): (name: keyof U) => boolean;
+  <K extends keyof U, U extends Record<PropertyKey, any>>(name: K, obj: U): boolean;
 };
-export function propEq<T, K extends PropertyKey>(val: T, name: K): (obj: Record<K, T>) => boolean;
-export function propEq<K extends keyof U, U>(val: U[K], name: K, obj: U): boolean;
+export function propEq<K extends PropertyKey>(__: Placeholder, name: K): {
+  (__: Placeholder, obj: Record<K, any>): <V>(val: V) => boolean;
+  <V>(val: V, obj: Record<K, any>): boolean
+  <V>(val: V): (obj: Record<K, any>) => boolean
+};
+export function propEq<V, K extends PropertyKey>(val: V, name: K): (obj: Record<K, any>) => boolean;
+
+
+export function propEq<U extends Record<any, any>>(__: Placeholder, ___: Placeholder, obj: U): {
+  (__: Placeholder, name: keyof U): {
+    (__: Placeholder): never;
+    <V>(val: V): boolean;
+  }
+  <V>(val: V, name: keyof U): boolean;
+  (__: Placeholder): never;
+  <V>(val: V): (name: keyof U) => boolean;
+};
```

Comment:
Overload order + `WidenLiterals<>`
```suggestion
export function propEq<U extends Record<PropertyKey, any>>(__: Placeholder, ___: Placeholder, obj: U): {
  <V>(val: V): (name: keyof U) => boolean;
  <K extends keyof U>(__: Placeholder, name: K): (val: WidenLiterals<U[K]>) => boolean;
  <K extends keyof U>(val: WidenLiterals<U[K]>, name: K): boolean;
};
```